.github/workflows/terraform_integration.yaml

```
@@ -10,13 +10,15 @@ on:
 
 jobs:
   tf_integration:
-    name: 'Terraform integration'
+    name: 'Terraform integration (on tf ${{ matrix.terraform_version }})'
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        terraform_version: [ '~1.4.0', '~1.5.0', '~1.6.0', '~1.7.0', 'latest' ] # haven't seen < 1.4 in the wild, so limit for sanity
```

As this is used across the three action definitions, can we put it as a `repository variable` in settings? 
Creating a variable named `TERRAFORM_VERSIONS` with the value of `[ '~1.4.0', '~1.5.0', '~1.6.0', '~1.7.0', 'latest' ]`
and then accessing in the yaml with 

```suggestion
        terraform_version: ${{ fromJSON(vars.TERRAFORM_VERSIONS) }} # haven't seen < 1.4 in the wild, so limit for sanity
```

---

Ah, please skip the comment but I'd leave it as a reference. I see that each actions have a different set of versions

---

yeah, certainly there's a lot of duplication across the workflows. My bigger concern is that these workflows are 90% the same across aws/gcp/azure

---

Then, maybe we can use `organization variables` and create a var for each action, so it can be reused across aws/gcp/azure and proxy too